core/src/scene/dataLayer.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include "scene/sceneLayer.h"
+#include <string>
+
+namespace Tangram {
+
+// DataLayer represents a top-level layer in the stylesheet, distinct from
+// SceneLayer by its association with a collection within a DataSource
+class DataLayer {
```

Comment:
Forget my last comment, my head was in Java land. Does inheritance add
anything here? (Aside from more virtual dispatch at runtime ;) )
On Aug 8, 2015 8:09 PM, "Varun" <notifications@github.com> wrote:

> In core/src/scene/dataLayer.h
> <https://github.com/tangrams/tangram-es/pull/237#discussion_r36583068>:
>
> > @@ -0,0 +1,26 @@
> > +#pragma once
> > +
> > +#include "scene/sceneLayer.h"
> > +#include <string>
> > +
> > +namespace Tangram {
> > +
> > +// DataLayer represents a top-level layer in the stylesheet, distinct from
> > +// SceneLayer by its association with a collection within a DataSource
> > +class DataLayer {
>
> Hmm.. I am not sure I get this.
>
> How in case of inheritance "DataLayer" will container other "DataLayer"s?
>
> DataLayer being a specialized SceneLayer will still have vector of
> SceneLayers (subLayers).
> To me inheritance makes more sense.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tangrams/tangram-es/pull/237/files#r36583068>.
>